src/main/java/net/sf/mzmine/modules/peaklistmethods/io/gnpsexport/GNPSExportTask.java

```diff
@@ -212,10 +226,23 @@ private void export(PeakList peakList, FileWriter writer, File curFile) throws I
         writer.write("CHARGE=" + msmsCharge + msmsPolarity + newLine);
 
         writer.write("MSLEVEL=2" + newLine);
-
-        DataPoint peaks[] = massList.getDataPoints();
-        for (DataPoint peak : peaks) {
-          writer.write(peak.getMZ() + " " + peak.getIntensity() + newLine);
+        boolean merged = false;
+        if (merger != null) {
+          List<MergedSpectrum> spectrum = merger.merge(row, massListName);
+          Optional<MergedSpectrum> max = spectrum.stream().max((u, v) -> Double.compare(v.getBestFragmentScanScore(), u.getBestFragmentScanScore()));
+          if (max.isPresent() && max.get().data.length>0) {
+            for (DataPoint peak : max.get().data) {
+              writer.write(peak.getMZ() + " " + peak.getIntensity() + newLine);
```

Comment:
This line is repeated for both the merged and unmerged blocks... Perhaps it would make more sense to assign a variable (like ```MassSpectrum spectrumToExport```) and then send that one to the writer. If merging, spectrumToExport can point to the merged spectrum, otherwise to the most intense one.

Reply:
done